modules/hal_bouffalolab/CMakeLists.txt

```diff
@@ -1,61 +1,15 @@
 # Copyright (c) 2021-2024 Gerson Fernando Budke <nandojve@gmail.com>
+# Copyright (c) 2024 MASSDRIVER EI (massdriver.space)
 #
 # SPDX-License-Identifier: Apache-2.0
 
 if(CONFIG_SOC_FAMILY_BFLB)
 
-zephyr_library_named(hal_bouffalolab)
-
-zephyr_compile_definitions(
-  ARCH_RISCV
-)
-
-zephyr_library_compile_definitions(
-  BFLB_USE_HAL_DRIVER
-  BFLB_USE_CUSTOM_LD_SECTIONS
-)
-
-set(bflb_soc         ${CONFIG_SOC_SERIES}02)
-set(bflb_drv_dir     ${ZEPHYR_HAL_BOUFFALOLAB_MODULE_DIR}/drivers/${bflb_soc}_driver)
-set(bflb_common_dir  ${ZEPHYR_HAL_BOUFFALOLAB_MODULE_DIR}/common)
-set(bflb_drv_src_dir ${bflb_drv_dir}/std_drv/src)
-
-# Global includes to be used outside hal_gigadevice
-zephyr_include_directories(
-  include
-  ${ZEPHYR_HAL_BOUFFALOLAB_MODULE_DIR}/include
-
-  ${bflb_drv_dir}/regs
-  ${bflb_drv_dir}/startup
-  ${bflb_drv_dir}/std_drv/inc
-
-  ${bflb_common_dir}/misc
-)
-
-zephyr_library_include_directories(
-  ${bflb_common_dir}/soft_crc
-)
-
-zephyr_library_sources(
-  ${bflb_drv_src_dir}/${bflb_soc}_aon.c
-  ${bflb_drv_src_dir}/${bflb_soc}_ef_ctrl.c
-  ${bflb_drv_src_dir}/${bflb_soc}_glb.c
-  ${bflb_drv_src_dir}/${bflb_soc}_hbn.c
```

Comment:
no, this needs to be squashed, this whole thing is a mess, this goes directly against the recent architecture working group decision of completely changing things you just added in the same PR